src/lib/common/sol-platform-impl-linux-micro.c

```diff
@@ -514,16 +521,108 @@ teardown_pid1(void)
     reboot(reboot_cmd);
 }
 
+static bool
+sol_platform_linux_micro_should_debug(struct gdbserver *gdb)
+{
+    int idx, argc = sol_argc();
+    char **argv = sol_argv();
+    bool res = false;
+
+    for (idx = 1; idx < argc; idx++) {
```

Comment:
No, I haven't tested as PID1 yet, I just read the kernel source code and it will pass the "after dash" args as argv.... anyway, it's worth testing and make sure it works somehow.